Cabal/src/Distribution/Simple/GHC/Internal.hs

```diff
@@ -576,6 +578,7 @@ componentGhcOptions verbosity implInfo lbi bi clbi odir =
     , ghcOptFfiIncludes = toNubListR $ includes bi
     , ghcOptObjDir = toFlag odir
     , ghcOptHiDir = toFlag odir
+    , ghcOptHieDir = bool NoFlag (toFlag $ odir </> extraCompilationArtifacts) $ flagHie implInfo
```

Comment:
It seems a bit unfortunate to be placing files directly in `extra-compilation-artifacts`. My hope when we agreed on this design was that users would instead place things in distinct subdirectories to avoid clutter and the potential for name conflicts. I wonder if it wouldn't be better to rather place these in `odir </> extraCompilationArtifacts </> "hie"` or similar.

Reply:
That seems reasonable to me. I will get a PR going for that, probably on Monday.

Reply:
Addressed by #9244